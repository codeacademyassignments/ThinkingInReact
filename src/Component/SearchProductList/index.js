import React, {Component} from 'react';
import ProductTable from '../ProductTable'
import SearchBar from '../SearchBar';

class SearchProductList extends Component {

    constructor(props){
      super(props);
      this.state={
        filterText: '',
        inStockOnly: false,
      }
    }
    onTextChange=(key)=>{//arrow function automatically binds the this instance
      this.setState({
        filterText:key,
      })

    }
    onTickChange=(inStockOnly)=>{
      console.log(inStockOnly);
      this.setState({
        inStockOnly:inStockOnly,
      })
     
    }
    render(){

      return (
        <div style={{textAlign:'center'}}>
        <div style={{display:'inline-block',marginTop:'50px'}}>
          <SearchBar onTextChange={this.onTextChange} onTickChange={this.onTickChange} filterText={this.state.filterText} inStockOnly={this.state.inStockOnly}/>
          <ProductTable inStockOnly={this.state.inStockOnly} searchKey={this.state.filterText} products={this.props.products}/>
        </div>
        </div>

      )
    }
}

export default SearchProductList;