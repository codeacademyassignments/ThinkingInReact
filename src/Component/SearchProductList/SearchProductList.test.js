import './index.js';
import renderer from 'react-test-renderer';
import SearchProductList from './index';
import React from 'react';
import {shallow} from 'enzyme';


const products = [
  {category: "Sporting Goods", price: "$29.99", stocked: false, name: "Basketball"},
  {category: "Electronics", price: "$99.99", stocked: true, name: "iPod Touch"},
  {category: "Electronics", price: "$399.99", stocked: false, name: "iPhone 5"},
  {category: "Sporting Goods", price: "$49.99", stocked: true, name: "Football"},
  {category: "Electronics", price: "$199.99", stocked: true, name: "Nexus 7"},
  {category: "Sporting Goods", price: "$9.99", stocked: true, name: "Baseball"} 
];
describe('snapshot testing',()=>{

  const tree = renderer.create(<SearchProductList products={products}/>);
  expect(tree).toMatchSnapshot();
})

describe('onTextChange',()=>{
  const wrapper = shallow(<SearchProductList products={products}/>);
  it('should change the filterText property of state to the string passed',()=>{
    wrapper.instance().onTextChange('foot');
    expect(wrapper.instance().state.filterText).toEqual('foot');
  })
  it('should change the filterText property of state to the integer passed',()=>{
    wrapper.instance().onTextChange(123);
    expect(wrapper.instance().state.filterText).toEqual(123);
  })
  it('should change the filterText property of state to the empty string when empty string passed',()=>{
    wrapper.instance().onTextChange('');
    expect(wrapper.instance().state.filterText).toEqual('');
  })
})

describe('onTickChange',()=>{
  const wrapper = shallow(<SearchProductList products={products}/>);
  it('should change the Tick property to the value passed',()=>{
    wrapper.instance().onTickChange(true);
    expect(wrapper.instance().state.inStockOnly).toEqual(true);
  })
})