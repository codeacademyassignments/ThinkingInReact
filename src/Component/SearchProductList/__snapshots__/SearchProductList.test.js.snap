// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    style={
      Object {
        "display": "inline-block",
        "marginTop": "50px",
      }
    }
  >
    <div>
      <input
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
      <br />
      <input
        onClick={[Function]}
        type="checkbox"
      />
      Only show products in stock
    </div>
    <table>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Price
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            colSpan="2"
          >
            Electronics
          </th>
        </tr>
        <tr>
          <td>
            iPod Touch
          </td>
          <td>
             
            $99.99
          </td>
        </tr>
        <tr>
          <td>
            <span
              id="name"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              iPhone 5
            </span>
          </td>
          <td>
             
            $399.99
          </td>
        </tr>
        <tr>
          <td>
            Nexus 7
          </td>
          <td>
             
            $199.99
          </td>
        </tr>
        <tr>
          <th
            colSpan="2"
          >
            Sporting Goods
          </th>
        </tr>
        <tr>
          <td>
            <span
              id="name"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              Basketball
            </span>
          </td>
          <td>
             
            $29.99
          </td>
        </tr>
        <tr>
          <td>
            Football
          </td>
          <td>
             
            $49.99
          </td>
        </tr>
        <tr>
          <td>
            Baseball
          </td>
          <td>
             
            $9.99
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
